tests/integration/targets/aci_access_spine_switch_selector/tasks/main.yml

```diff
@@ -0,0 +1,162 @@
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact: 
+   aci_info: &aci_info
+    host: "{{ aci_hostname }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    validate_certs: "{{ aci_validate_certs | default(false) }}"
+    use_ssl: "{{ aci_use_ssl | default(true) }}"
+    use_proxy: "{{ aci_use_proxy | default(true) }}"
+    output_level: "{{ aci_output_level | default('info') }}"
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Deleting Switch Policy Spine profile exists for kick off
+    cisco.aci.aci_access_spine_switch_profile:
+      <<: *aci_info
+      spine_profile: sw_name_test
+      state: absent
+
+  - name: Ensuring Switch Policy Spine profile exists for kick off
+    cisco.aci.aci_access_spine_switch_profile: &aci_access_spine_switch_profile_present
+      <<: *aci_info
+      spine_profile: sw_name_test
+      state: present
+    register: spine_profile_present
+
+  # TODO: Ensure that spine Policy Group Exists (module missing) (infra:AccPortGrp)
+
+  - name: Adding a switch policy spine profile selector associated Node Block range (w/o policy group) - check mode works
+    cisco.aci.aci_access_spine_switch_selector: &aci_access_spine_switch_selector_present
+      <<: *aci_access_spine_switch_profile_present
+      spine: spine_selector_name
+      spine_node_blk: node_blk_name
+      from: 1011
+      to: 1011
+    check_mode: yes
+    register: sw_spine_selec_check_mode_present
+
+  - name: Adding a switch policy spine profile selector associated Node Block range (w/o policy group) - creation works
+    cisco.aci.aci_access_spine_switch_selector:
+      <<: *aci_access_spine_switch_selector_present
+    register: sw_spine_selec_present
+
+  - name: Adding a switch policy spine profile selector associated Node Block range (w/o policy group) - idempotency works
+    cisco.aci.aci_access_spine_switch_selector:
+      <<: *aci_access_spine_switch_selector_present
+    register: sw_spine_selec_idempotent
+
+  - name: Adding a switch policy spine profile selector associated Node Block range (w/ policy group) - update works
+    cisco.aci.aci_access_spine_switch_selector:
+      <<: *aci_access_spine_switch_selector_present
+      policy_group: anstest_policygroupname
+    register: sw_spine_selec_update
+
+  # TODO: also test for errors
```

Comment:
Was there meant to be tests for errors or is this a note for later PRs?

Reply:
notes I forgot to remove. I removed all unnecessary ones. Thank you